app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -775,4 +775,34 @@ class AppointmentRepositoryAndroidTest {
     val (appointment) = appointmentRepository.lastCreatedAppointmentForPatient(patientId).blockingFirst()
     assertThat(appointment!!).isEqualTo(secondAppointment)
   }
+
+  @Test
+  fun marking_appointment_older_than_current_date_should_work_correctly() {
+    val patientId = UUID.randomUUID()
+    val scheduleDate = LocalDate.now(testClock).plusMonths(1)
+
+    testClock.advanceBy(Duration.ofHours(1))
+
+    val firstAppointment = appointmentRepository.schedule(patientId, scheduleDate).blockingGet()
+
+    testClock.advanceBy(Duration.ofHours(1))
+
+    appointmentRepository.markAppointmentsCreatedBeforeTodayAsVisited(patientId).blockingAwait()
+
+    val (lastAppointment) = appointmentRepository.lastCreatedAppointmentForPatient(patientId).blockingFirst()
+    assertThat(lastAppointment).isEqualTo(firstAppointment)
+
+    testClock.advanceBy(Duration.ofDays(1))
+
+    appointmentRepository.schedule(patientId, scheduleDate).blockingGet()
```

Comment:
`AppointmentRepository#schedule` internally marks all older appointments as visited, so this test does not verify whether `markAppointmentCreatedBeforeTodayAsVisited` works as expected.

Reply:
Ideally, when testing a class, we should not invoke other methods in the class except the one that we are testing and use other means to verify the method behaviour.

Reply:
Good catch. I will use the `dao` directly.